cryptography/src/bls12381/dkg.rs

```diff
@@ -1047,9 +1053,16 @@ impl<V: Variant, S: Signer> Dealer<V, S> {
     ///
     /// This should be called at the point where no more acks will be processed.
     pub fn finalize(self) -> SignedDealerLog<V, S> {
```

Comment:
Should we just return `Option<>` here if we don't have any results to share when finalize? Feels weird to return a log with no results?

Reply:
I don't like having this be optional, because it introduces a separate bit of control flow in the resulting application, because you need to distinguish betweeen a dealer not yet having posted their outcome, and them posting an outcome which is empty.

Maybe we should have the `SignedDealerLog` have a signature over `Option<DealerLog>` instead. That way it becomes clear that the dealer did post a result, that we have a guarantee that this dealer actually authorized that result, but that they don't have a log.

Reply:
Decided to make this flow a bit more explicit by introducing an enum.